### DISCUSSION TITLE
Creating an app client for a NestJs backend service - better to use Confidential Client or Other?

### DISCUSSION THREAD

**Discussion by mtander** [Lokicoule/nestjs-cognito#952] [Q&A]
I am unsure of which type of App Client to create in my user pool for a backend service.  I spent some time reading AWS documentation and other online guides, and I read somewhere that NestJs is not compatible with using a client secret, but I don't know if that's true or not.  Are there specific guidelines for how to build the app client in a way that will work with this package?

Thank you,

Mary

**Comment by Lokicoule** [comment]
Hey Mary,

I want to confirm that NestJs does not inherently limit the use of client secrets. However, if your concerns involve rotating client secrets, there may be limitations due to the specific way `@nestjs-cognito` handles them.

To provide you with the most accurate advice, I need a bit more detail about your backend service and how it interacts with AWS Cognito. Could you share more about the functionalities your service needs to perform and any specific security requirements you have?
